src/wallet/wallet.h

```diff
@@ -732,14 +732,11 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
      */
     mutable RecursiveMutex cs_wallet;
 
-    /** Get database handle used by this wallet. Ideally this function would
-     * not be necessary.
-     */
-    WalletDatabase& GetDBHandle()
+    WalletDatabase& GetDatabase() const override
     {
-        return *database;
+        assert(static_cast<bool>(m_database));
```

Comment:
@ryanofsky made this change in the context of https://github.com/bitcoin-core/gui/issues/95#issuecomment-694254368, where `m_database` can be null.